### PR TITLE
consumer forms selection 

### PR DESCRIPTION
**Pull request by sebastiankb**
also see #877

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/pull/1108.html" title="Last updated on Apr 28, 2021, 12:39 PM UTC (00336e1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/1108/24c2bf8...00336e1.html" title="Last updated on Apr 28, 2021, 12:39 PM UTC (00336e1)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -2096,7 +2096,12 @@ <h2>Hypermedia Controls Vocabulary Definitions</h2>
           "rfc2119" title="MUST">MUST</em> include an entry in the array
           associated with the name <code>additionalResponses</code> that
           includes a value for the name <code>schema</code>.</span> 
-          </p></section>
+          </p><p>
+          A <a>Thing Description</a> of a WoT producer may has multiple forms entries with, e.g., different
```
Comment (zolkis) [inline index.html:2100]: nit: may have

File: index.html [hunk 2]
```diff
@@ -2096,7 +2096,12 @@ <h2>Hypermedia Controls Vocabulary Definitions</h2>
           "rfc2119" title="MUST">MUST</em> include an entry in the array
           associated with the name <code>additionalResponses</code> that
           includes a value for the name <code>schema</code>.</span> 
-          </p></section>
+          </p><p>
+          A <a>Thing Description</a> of a WoT producer may has multiple forms entries with, e.g., different 
+          protocol and/or content types declarations that a <a>Consumer</a> would also support. In that case
```
Comment (zolkis) [inline index.html:2101]: "could possibly support" (may be more clear to me, but you decide)

File: index.html [hunk 3]
```diff
@@ -2096,7 +2096,12 @@ <h2>Hypermedia Controls Vocabulary Definitions</h2>
...
+          it is recommended that the <a>Consumer</a> chooses the first form entry that works for them.
```
Comment (zolkis) [inline index.html:2102]: I think this should be formulated in more generic terms, something like "In that case the <a>Consumer</a> may choose any form entry that works for them".

File: validation/td-validation.ttl [hunk 4]
```diff
@@ -726,7 +726,12 @@
           "rfc2119" title="MUST">MUST</em> include an entry in the array
           associated with the name <code>additionalResponses</code> that
           includes a value for the name <code>schema</code>.</span> 
-          </p>"""^^rdf:HTML .
+          </p><p>
+          A <a>Thing Description</a> of a WoT producer may has multiple forms entries with, e.g., different
```
Comment (zolkis) [inline validation/td-validation.ttl:730]: nit: may have

File: validation/td-validation.ttl [hunk 5]
```diff
@@ -726,7 +726,12 @@
           "rfc2119" title="MUST">MUST</em> include an entry in the array
           associated with the name <code>additionalResponses</code> that
           includes a value for the name <code>schema</code>.</span> 
-          </p>"""^^rdf:HTML .
+          </p><p>
+          A <a>Thing Description</a> of a WoT producer may has multiple forms entries with, e.g., different 
+          protocol and/or content types declarations that a <a>Consumer</a> would also support. In that case
+          it is recommended that the <a>Consumer</a> chooses the first form entry that works for them.
+          </p>
```
Comment (zolkis) [inline validation/td-validation.ttl:733]: ah, copy-paste. Then same comments as above :)

**Review by zolkis — APPROVED** [review]
Approved with nits and alternative formulations to consider.

**Comment by egekorkan** [comment]
I agree with @zolkis comments, otherwise good :)

**Review by danielpeintner — CHANGES_REQUESTED** [review]
LGTM besides comments from @zolkis

File: index.html (hunk 3 above)
Comment (relu91) [reply index.html:2102]: Should we also suggest a retaining policy? For example "When one form is chosen, it is expected that the consumer will continue to use it as long as possible for every new interaction with the Producer" ? 🤔  

Also _works for them_ feels a little bit vague but I understand that we might not need to be specific about implementation details.  What happens if the form chosen is unreachable? The consumer should give up or try again with another form?

**Review by relu91 — APPROVED** [review]
Just a little comment more than what @zolkis  already said.

File: index.html (hunk 3 above)
Comment (sebastiankb) [reply index.html:2102]: @zolkis 
>I think this should be formulated in more generic terms, something like "In that case the Consumer may choose any form entry that works for them".

I thought the idea was to provide a proposal, which one the Consumer should select if there are multiple possible.

@relu91 
>"When one form is chosen, it is expected that the consumer will continue to use it as long as possible for every new interaction with the Producer"

+1

File: index.html (hunk 1 above)
Comment (sebastiankb) [reply index.html:2100]: done

File: index.html (hunk 2 above)
Comment (sebastiankb) [reply index.html:2101]: done

File: index.html (hunk 3 above)
Comment (zolkis) [reply index.html:2102]: Yes, but "the first form entry that works for them" is equivalent to "any form entry".
Why is it a problem if the implementation chooses the form? (the one best matching its own context).
If we have generic guidelines for specific cases, we should be explicit about them, like when choosing between protocols, which would need defining a priority list for protocols (just an example).

File: index.html (hunk 1 above)
Comment (zolkis) [reply index.html:2100]: Still looks the same, don't forget you need to fix these comments in 2 places (copy-paste) - and I see you fixed it one place, not here. You can amend the previous commit I guess.

Comment (sebastiankb) [reply index.html:2100]: sorry, I forgot the render script

**Comment by sebastiankb** [comment]
updated PR based on feedback from @zolkis and @RealJoshue108

**Review by zolkis — APPROVED** [review]
Looks good.

The language is not concise, but that needs to be checked across the whole document.
For instance, instead of "it is expected" it would be nice to specify MAY (free to do or not to do) or SHOULD (recommended in all cases minus specific exceptions) or MUST (it is a compliance error not to do so).

**Comment by sebastiankb** [comment]
from today's TD call we decided to include this PR

**Pull request merged**